src/organizations/organization.service.ts

```diff
@@ -173,21 +183,23 @@ export class OrganizationService {
       }
 
       if (invitation.invitationRole == 'admin') {
-        // Check if the user is already an admin of the organization
         if (organization.roles[0].members.includes(invitee)) {
```

Comment:
I would not assume ordering of the roles. You need to find the role with name `admin` from `organization.roles`. Same for other occurences